stan/math/opencl/copy.hpp

```diff
@@ -89,6 +91,69 @@ void copy(Eigen::Matrix<double, R, C>& dst, const matrix_cl& src) {
   }
 }
 
+/**
+ * Packs the flat triagnular matrix on the OpenCL device and
+ * copies it to the std::vector.
+ *
+ * @tparam triangular_view the triangularity of the source matrix
+ * @param dst the destination std::vector
+ * @param src the flat triangular source matrix on the OpenCL device
+ */
+template <TriangularViewCL triangular_view>
+inline void packed_copy(std::vector<double>& dst, const matrix_cl& src) {
```

Comment:
I used std::vector everywhere for the packed version. Should we use Eigen::VectorXd or something else?

Reply:
I think `std::vector` is faster unless you need any of Eigen's features.

Reply:
Or unless you're converting from Eigen types on the way in - assuming we're not expecting to do that right?

Reply:
Not really, this is intended primarily to be used to copy the vectors of `varis` as in the reverse mode cholesky decompose case.